apps/glusterfs/placer_arbiter.go

```diff
@@ -231,7 +231,11 @@ func (bp *ArbiterBrickPlacer) tryPlaceBrickOnDevice(
 
 	logger.Debug("Trying to place brick on device %v", device.Info.Id)
 
-	for _, b := range bs.Bricks {
+	for i, b := range bs.Bricks {
+		// skip the brick at the current index (in case this is a replace)
```

Comment:
I don't seem to understand this commit. Is this making arbiter placer different than standard brick placer? If yes, why?

Reply:
This change tries to get the arbiter placer to behave more like standard placer. In the current, non-arbiter, approach a series of nodes and with bricks like n1/d1 n1/d2 n2/d1 n2/d2 (and so on) you can potentially replace a brick on n1/d1 with one on n1/d2. However, my initial arbiter could would have rejected n1/d2 because the old brick is still visible in the brick set and so it would see n1 in use and block it.

In current non-arbiter code the existing brick is filtered out from the brick set. We can do that easily there because the position of the bricks in the brick sets "doesn't matter".

I'll try to quickly improve the comment a bit.